thunx-predicates-querydsl/src/main/java/com/contentgrid/thunx/predicates/querydsl/QueryDslConverter.java

```diff
@@ -38,33 +38,52 @@ public Expression<?> visit(FunctionExpression<?> function) {
 
         // convert all the terms
         var terms = function.getTerms().stream()
-                .map(term -> term.accept(this));
+                .map(term -> term.accept(this))
+                .collect(Collectors.toList());
 
         switch (function.getOperator()) {
             case EQUALS:
-                var equalTerms = terms
-                        .collect(Collectors.toList());
-                if(equalTerms.size() != 2) {
-                    throw new IllegalArgumentException("Equal operation requires 2 parameters.");
-                }
-                return ExpressionUtils.eq((Expression<Object>)equalTerms.get(0), equalTerms.get(1));
+                assertTwoTerms(terms);
+                return ExpressionUtils.eq((Expression<Object>) terms.get(0), terms.get(1));
+            case NOT_EQUAL_TO:
+                assertTwoTerms(terms);
+                return ExpressionUtils.ne(terms.get(0), (Expression<Object>) terms.get(1));
+            case GREATER_THAN_OR_EQUAL_TO:
+                assertTwoTerms(terms);
+                return Expressions.booleanOperation(Ops.GOE, terms.toArray(new Expression[0]));
+            case GREATER_THAN:
+                assertTwoTerms(terms);
+                return Expressions.booleanOperation(Ops.GT, terms.toArray(new Expression[0]));
+            case LESS_THEN_OR_EQUAL_TO:
+                assertTwoTerms(terms);
+                return Expressions.booleanOperation(Ops.LOE, terms.toArray(new Expression[0]));
+            case LESS_THAN:
+                assertTwoTerms(terms);
+                return Expressions.booleanOperation(Ops.LT, terms.toArray(new Expression[0]));
             case OR:
-                return ExpressionUtils.anyOf(terms.map(term -> (Predicate)term).collect(Collectors.toList()));
+                return ExpressionUtils.anyOf(terms.stream().map(term -> (Predicate) term).collect(Collectors.toList()));
             case AND:
-                return ExpressionUtils.allOf(terms.map(term -> (Predicate)term).collect(Collectors.toList()));
+                return ExpressionUtils.allOf(terms.stream().map(term -> (Predicate) term).collect(Collectors.toList()));
             default:
                 throw new UnsupportedOperationException(
                         "Operation '" + function.getOperator() + "' not implemented");
         }
     }
 
+    private static void assertTwoTerms(List<? extends Expression<?>> terms) {
+        if (terms.size() != 2) {
+            throw new IllegalArgumentException("Equal operation requires 2 parameters.");
```

Comment:
```suggestion
            throw new IllegalArgumentException("Operation requires 2 parameters.");
```

This is not only about the equals operator